src/Pangraph/GraphML/Parser.hs

```diff
@@ -1,28 +1,24 @@
-module Pangraph.GraphML.Parser(
-parseGraphToAlga,
-parseTemplateToPangraph,
-template,
-PT.Template
+module Pangraph.GraphML.Parser (
+graphmlToPangraph,
+graphmlToPangraph'
 ) where
 
 import qualified Pangraph as P
 import qualified Text.XML.Hexml as H
 import qualified Data.ByteString as BS
 import qualified Pangraph.XMLTemplate as PT
-import qualified Algebra.Graph.HigherKinded.Class as H
 
-
-parseGraphToAlga:: (H.Graph g) => PT.Template -> BS.ByteString -> Either BS.ByteString (g P.Node)
-parseGraphToAlga t file =
-  case H.parse file of
-    Left x -> Left x
-    Right x -> Right $ PT.parseTemplateToAlga t x
-
-parseTemplateToPangraph :: PT.Template -> BS.ByteString -> Either BS.ByteString P.Pangraph
-parseTemplateToPangraph t file =
+graphmlToPangraph :: BS.ByteString -> Either BS.ByteString (Either [P.MalformedEdge] P.Pangraph)
+graphmlToPangraph file =
   case H.parse file of
     Left x -> Left x
-    Right x -> Right $ PT.parseTemplateToPangraph t x
+    Right x -> Right $ PT.hexmlToPangraph PT.graphMLTemplate x
 
-template:: [PT.Template]
-template = PT.graphMLTemplate
+graphmlToPangraph' :: BS.ByteString -> P.Pangraph
```

Comment:
This is an unsafe function, and it's useful to mark such functions with the `unsafe` prefix.

Let's rename it to `unsafeGraphmlToPangraph`. Or, even better, to `unsafeParse`, because the return type (`Pangraph`) is clear from the type signature, and the `GraphML` should be the name of the module instead.